pkg/asset/store/store.go

```diff
@@ -357,3 +372,27 @@ func (s *storeImpl) purge(excluded []asset.WritableAsset) error {
 func increaseIndent(indent string) string {
 	return indent + "  "
 }
+
+// configDriveImage create config drive iso with ignition-config file
+// refer to https://coreos.com/os/docs/latest/config-drive.html
+func configDriveImage(file string) error {
+	logrus.Infof("Creating config drive image for %s", file)
+	if err := os.MkdirAll("openstack/latest", 0755); err != nil {
+		return err
+	}
+	cmd := exec.Command("cp", file, "openstack/latest/user_data")
+	_, err := cmd.Output()
+	if err != nil {
+		return err
+	}
+	cmd = exec.Command("mkisofs", "-R", "-V", "config-2", "-o", "configdrive.iso", ".")
```

Comment:
I don't think you need to bother with ISOs. Ignition is just looking for `/dev/disk/by-label/config-2`, so any filesystem with that label will work.

Reply:
@crawford Do you mean that the suffix ".iso" is not needed? I'll have a try.

Reply:
openstack itself only support iso or vfat.. but seems the ignition doesn't have this restriction
https://github.com/coreos/ignition/blob/master/internal/providers/openstack/openstack.go#L117
so we should be able to avoid iso from code here

Reply:
> openstack itself only support iso or vfat.. but seems the ignition doesn't have this restriction
> https://github.com/coreos/ignition/blob/master/internal/providers/openstack/openstack.go#L117
> so we should be able to avoid iso from code here

@jichenjc  Yes, I agree.

Reply:
> I don't think you need to bother with ISOs. Ignition is just looking for `/dev/disk/by-label/config-2`, so any filesystem with that label will work.

@crawford The ignition reads the filesystem's [CONFIG-2/config-2 label](https://github.com/coreos/ignition/blob/a924dd7d11685a30bf79f20ce9fc3ca93860f032/internal/providers/openstack/openstack.go#L75), how can we make a label to a file without ISOs. 

I've found a method like below and I'll have a try. https://docs.openstack.org/ironic/latest/install/configdrive.html#accessing-the-configuration-drive-data
```
mkdir -p /openstack/latest/user_data
mount /dev/disk/by-label/config-2 /openstack/latest/user_data
```

Reply:
udev populates those symlinks based on the filesystem label. All you need is a filesystem which supports labelling (e.g. ext4, xfs, vfat).

> how can we make a label to a file without ISOs

You can't - at least, not in the initramfs. You'll need to rely on the mechanism mentioned above to get the config to Ignition.

I originally made the suggestion in the hopes that it might simplify the implementation and maybe drop some dependencies (I was hoping someone had put together a Golang implementation for FAT that we could just use to create the filesystem ourselves). If this alternate route ends up being more complicated, then I would abandon it. We'll need to make sure that the `mkisofs` utility is available in the test environment and the MAO container image, if we continue down this route.

Reply:
right, avoid the utils will be a good thing, do we have a fixed format for our disks ? otherwise seems the ext2/3/4/ using e2progs and xfs using xfs_admin.. so maybe we need consider the logic here as well.. @gyohuangxin